desc/compute/_fast_ion.py

```diff
@@ -0,0 +1,355 @@
+                points,
+                is_fourier=True,
+            )
+            # This is γ_c π/2.
+            gamma_c = jnp.arctan(
+                safediv(
+                    drift1,
+                    drift2
+                    * bounce.interp_to_argmin(
+                        data["|grad(rho)|*|e_alpha|r,p|"], points, is_fourier=True
+                    ),
+                )
+            )
+            return jnp.sum(v_tau * gamma_c**2, axis=-1)
+
+        return jnp.sum(
+            batch_map(fun, data["pitch_inv"], pitch_batch_size)
+            * data["pitch_inv weight"]
+            / data["pitch_inv"] ** 2,
+            axis=-1,
+        ) / (bounce.compute_fieldline_length(fieldline_quad) * 2**1.5 * jnp.pi)
+
+    grid = transforms["grid"]
+    # It is assumed the grid is sufficiently dense to reconstruct |B|,
+    # so anything smoother than |B| may be captured accurately as a single
+    # Fourier series rather than transforming each component.
+    # Last term in K behaves as ∂log(|B|²/B^ϕ)/∂ρ |B| if one ignores the issue
+    # of a log argument with units. Smoothness determined by positive lower bound
+    # of log argument, and hence behaves as ∂log(|B|)/∂ρ |B| = ∂|B|/∂ρ.
+    data["Gamma_c"] = _compute(
+        Gamma_c,
+        fun_data={
+            "|grad(psi)|*kappa_g": data["|grad(psi)|"] * data["kappa_g"],
+            "|grad(rho)|*|e_alpha|r,p|": data["|grad(rho)|"] * data["|e_alpha|r,p|"],
+            "|B|_r|v,p": data["|B|_r|v,p"],
+            "K": data["iota_r"]
+            * dot(cross(data["grad(psi)"], data["b"]), data["grad(phi)"])
+            - (
+                2 * data["|B|_r|v,p"]
+                - data["|B|"] * data["B^phi_r|v,p"] / data["B^phi"]
+            ),
+        },
+        data=data,
+        theta=theta,
+        grid=grid,
+        num_pitch=num_pitch,
+        simp=False,
+        surf_batch_size=surf_batch_size,
+    )
+    return data
+
+
+def _cvdrift0(data, B, pitch):
+    return safediv(
+        data["cvdrift0"] * (1 - 0.5 * pitch * B), jnp.sqrt(jnp.abs(1 - pitch * B))
+    )
+
+
+def _gbdrift(data, B, pitch):
+    return safediv(
+        (data["gbdrift (periodic)"] + data["gbdrift (secular)/phi"] * data["zeta"])
+        * (1 - 0.5 * pitch * B),
+        jnp.sqrt(jnp.abs(1 - pitch * B)),
+    )
+
+
+@register_compute_fun(
+    name="Gamma_c Velasco",
+    label=(
+        # Γ_c = π/(8√2) ∫ dλ 〈 ∑ⱼ [v τ γ_c²]ⱼ 〉
+        "\\Gamma_c = \\frac{\\pi}{8 \\sqrt{2}} "
+        "\\int d\\lambda \\langle \\sum_j (v \\tau \\gamma_c^2)_j \\rangle"
+    ),
+    units="~",
+    units_long="None",
+    description="Fast ion confinement proxy "
+    "as defined by Velasco et al. (doi:10.1088/1741-4326/ac2994)",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="r",
+    data=[
+        "min_tz |B|",
+        "max_tz |B|",
+        "cvdrift0",
+        "gbdrift (periodic)",
+        "gbdrift (secular)/phi",
```

Comment:
Should we add a test to ensure that the secular gbdrift gives the same answer as periodic gbdrift as field line length is increased? Otherwise, it can be added later.

Reply:
It will not because the secular term drives the computation here to zero. We warn this in the objective docstring for now.